### DISCUSSION TITLE
Shared map between analysis.Pass

### DISCUSSION THREAD

**Discussion by alexal** [golangci/golangci-lint#2253] [Q&A] [answered]
Hello,

I'm working on a linter, the validation code is working just fine, but I need to preload the configuration options (from flags) and create a map (based on flag values and external files) that will be shared between each `analysis.Pass`. What is the best approach to do that?

Thanks.

**Comment by butuzov** [comment, marked as answer]
Not sure if best, but i did instantiate struct, and 'run' was its method. First call of run checked flags with sync.Once.

**Reply by alexal** [reply]
@butuzov thank you, Oleg, that works just perfect.
